src/GovUk.Education.ExploreEducationStatistics.Common/Model/Either.cs

```diff
@@ -35,6 +35,16 @@ public Either(TR right)
 
         public Either<TL, T> OnSuccess<T>(Func<TR, T> func) => Map(func);
 
+        public async Task<Either<TL, T>> OnSuccess<T>(Func<Task<Either<TL, T>>> func)
```

Comment:
Haha, I think we've both added the same extension method in separate PRs - your test coverage is better though :)